miniapps/tools/convert-dc.cpp

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) 2010, Lawrence Livermore National Security, LLC. Produced at
+// the Lawrence Livermore National Laboratory. LLNL-CODE-443211. All Rights
+// reserved. See file COPYRIGHT for details.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability see http://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the GNU Lesser General Public License (as published by the Free
+// Software Foundation) version 2.1 dated February 1999.
+//
+//      ---------------------------------------------------------------
+//      Convert DC: Convert between different types of data collections
+//      ---------------------------------------------------------------
+//
+// This tool demonstrates how to convert between MFEM's different concrete
+// DataCollection options.
+//
+// Currently supported data collection type options:
+//    visit:                VisItDataCollection (default)
+//    sidre or sidre_hdf5:  SidreDataCollection
+//    json:                 ConduitDataCollection w/ protocol json
+//    conduit_json:         ConduitDataCollection w/ protocol conduit_json
+//    conduit_bin:          ConduitDataCollection w/ protocol conduit_bin
+//    conduit_hdf5:         ConduitDataCollection w/ protocol hdf5
+//
+// Compile with: make convert-dc
+//
+// Serial sample runs:
+//    convert-dc -s ../../examples/Example5 -st visit -o Example5_Conduit -ot json
+//
+// Parallel sample runs:
+//    mpirun -np 4 convert-dc -s ../../examples/Example5-Parallel -st visit
+//                            -o Example5-Parallel_Conduit -ot json
+
+#include "mfem.hpp"
+
+using namespace std;
+using namespace mfem;
+
+DataCollection *create_data_collection(const std::string &dc_name,
+                                       const std::string &dc_type)
+{
+   DataCollection *dc = NULL;
+
+   if (dc_type == "visit")
+   {
+#ifdef MFEM_USE_MPI
+      dc = new VisItDataCollection(MPI_COMM_WORLD, dc_name);
+#else
+      dc = new VisItDataCollection(dc_name);
+#endif
+   }
+   else if ( dc_type == "sidre" || dc_type == "sidre_hdf5")
+   {
+#ifdef MFEM_USE_SIDRE
+      dc = new SidreDataCollection(dc_name);
+#else
+      MFEM_ABORT("Must build with MFEM_USE_SIDRE=YES for sidre support.");
+#endif
+   }
+   else if ( dc_type == "json" ||
+             dc_type == "conduit_json" ||
+             dc_type == "conduit_bin"  ||
+             dc_type == "hdf5")
```

Comment:
Shouldn't this be "conduit_hdf5"?

Reply:
@v-dobrev good catch, I had it in-correct in the comments and command line docs above. 
`hdf5` is the expected protocol name, I changed the comments and docs to `hdf5` instead of `conduit_hdf5`